src/hotspot/cpu/x86/macroAssembler_x86_pow.cpp

```diff
@@ -2562,6 +2566,16 @@ void MacroAssembler::fast_pow(XMMRegister xmm0, XMMRegister xmm1, XMMRegister xm
   jmp(L_2TAG_PACKET_21_0_2);
 
   bind(L_NOT_DOUBLE2);
+  // Special case: pow(x, 0.5) => sqrt(x)
+  ucomisd(xmm1, ExternalAddress(DOUBLE0DOT5)); // For pow(x, y), check whether y == 0.5
+  jccb(Assembler::notEqual, L_NOT_DOUBLE0DOT5);
+  jccb(Assembler::parity, L_NOT_DOUBLE0DOT5);
+  ucomisd(xmm0, ExternalAddress(DOUBLE0));
+  jccb(Assembler::belowEqual, L_NOT_DOUBLE0DOT5); // pow(x, 0.5) => sqrt(x) only for x > 0.0
```

Comment:
Add comment about why 0. case is skipped: -0.0 case.

Reply:
> Add comment about why 0. case is skipped: -0.0 case.

Thanks @vnkozlov for your review.

Comments have been added.
Thanks.